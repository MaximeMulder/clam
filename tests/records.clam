type A = {a: Int};
type B = {a: Int, b: String};

def main = {
    var a: A = { a = 0 };
    var b: B = { a = 1, b = "Hello" };

    var c: A = b;
    var d: A = { a = 1, b = "Hello" };

    print(a);
    print(b);

    print(a.a);
    print(b.a);
    print(b.b);
    unit
};
