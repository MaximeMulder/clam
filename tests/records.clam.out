{a = 0}
{a = 1, b = "Hello"}
0
1
"Hello"
